control-plane/subcommand/sync-catalog/command.go

```diff
@@ -393,6 +401,22 @@ func (c *Command) Run(args []string) int {
 	}
 }
 
+// remove all k8s services from Consul.
+func (c *Command) removeAllK8SServicesFromConsul(consulConfig *consul.Config) error {
+	consulClient, err := consul.NewClientFromConnMgr(consulConfig, c.connMgr)
+	if err != nil {
+		c.UI.Error(fmt.Sprintf("unable to create Consul API client: %s", err))
+		return err
+	}
+	_, err = consulClient.Catalog().Deregister(&api.CatalogDeregistration{Node: c.flagConsulNodeName}, nil)
```

Comment:
do we need to care about namespaces at all here?

Reply:
do we need to be concerned about the blast radius for deregistering the entire node here? will that affect the consul service registration?

Reply:
1. I distribute the de-register work into batches to avoid the traffic blaster
2. I added the `filter` flag to allow the user to control the de-register process in more granular level